runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5777,7 +5778,17 @@ class INTERPRETER_CLASS
 				omrthread_monitor_t monitor = getMonitorForWait(_currentThread, waitObject);
 				monitor->count = _currentThread->currentContinuation->waitingMonitorEnterCount;
 				_currentThread->currentContinuation->waitingMonitorEnterCount = 0;
-				returnVoidFromINL(REGISTER_ARGS, 4);
+				_currentThread->ownedMonitorCount -= 1;
+				if (J9VMJAVALANGTHREAD_DEADINTERRUPT(_currentThread, _currentThread->threadObject)) {
+					buildInternalNativeStackFrame(REGISTER_ARGS);
```

Comment:
A native stack frame was already built on line 5749. Do we need to build another native frame over here?

Reply:
yes, the one built on 5749 is for the carrier thread, once the stack is swapped, the native frame on the vthread is popped before reaching here.

Reply:
we should add the above justification as a comment here to allow other readers to easily understand this change.